web/pandas/pdeps/0010-required-pyarrow-dependency.md

```diff
@@ -0,0 +1,68 @@
+# PDEP-10: PyArrow as a required dependency
+
+- Created: 17 April 2023
+- Status: Under discussion
+- Discussion: [#52711](https://github.com/pandas-dev/pandas/pull/52711)
+              [#52509](https://github.com/pandas-dev/pandas/issues/52509)
+- Author: [Matthew Roeschke](https://github.com/mroeschke)
+- Revision: 1
+
+## Abstract
+
+This PDEP proposes that:
+
+- PyArrow becomes a runtime dependency starting with pandas 2.1
+- The minimum version of PyArrow supported starting with pandas 2.1 is version 7 of PyArrow.
+- The minimum version of PyArrow will be bumped every major pandas release to the highest
```

Comment:
Hmm.. This might be too aggressive and might also make it hard to predict what the minimum version will be.

I'd recommend following what we do for numpy, which is according to NEP 29, support
"all minor versions of NumPy released in the prior 24 months from the anticipated release date with a minimum of 3 minor versions of NumPy", for arrow as well.

Reply:
I think the challenge with offering a similar support window for the two libraries is that NumPy has a very stable ABI whereas PyArrow does not

Reply:
Not sure if I'm missing something, but sounds like what @lithomas1 is proposing is pretty much the same as what's written in the proposal but phrased in a different way. Has the proposal been updated, or am I misunderstanding that supporting the releases of the last 24 months, and supporting the highest/oldest version two years old?

Reply:
I read this as every major release we will bump the min required version of pyarrow to the latest version, but might be misreading here.


Note that my proposed change would be different in that we would drop Arrow versions in both major/minor versions (as opposed to every major version), just like we do with numpy (once we reach the end of the NEP support window). 

> I think the challenge with offering a similar support window for the two libraries is that NumPy has a very stable ABI whereas PyArrow does not

I might have missed some more discussion on this, but I thought we were going to restrict current usage of pyarrow to just what's exposed through Python.

Reply:
> I read this as every major release we will bump the min required version of pyarrow to the latest version, but might be misreading here.

To the latest version `that has been released for at least 2 years`. So, the minimum PyArrow version we support will be around 24 months old, and we should be supporting all the versions since that one, so more or less the same policy as NumPy. @mroeschke not sure if it's easy to rephrase in a way that it's more obvious what's the policy.

About bumping in major or minor releases, I don't have a preference, either is fine for me.

Reply:
Can you consider this to upgrading pyarrow in both major and minor versions to be consistent with numpy?

I ask because it is probably tricky for downstream to predict the length of our major release cycle (for 2.0 I think we delayed it twice. IIRC 1.4 was supposed to be 2.0).

Reply:
> Can you consider this to upgrading pyarrow in both major and minor versions to be consistent with numpy?

Sure that would be okay with me too